src/platform.ts

```diff
@@ -134,9 +183,8 @@ export class MideaPlatform implements DynamicPlatformPlugin {
         const accessory = new this.api.platformAccessory<MideaAccessory['context']>(device_info.name, uuid);
         const device = DeviceFactory.createDevice(this.log, device_info, this.platformConfig, deviceConfig);
         if (device) {
-          // We only need to login to Midea cloud if we are setting up a new accessory.  This is to
-          // retrieve token/key credentials.  If device was cached then we already have credentials.
           if (deviceConfig.advanced_options.token && deviceConfig.advanced_options.key) {
+            // token/key provided in config file, use those... set values in cached context
             this.log.info(`[${device_info.name}] New device at ${device_info.ip}:${device_info.port}, using token/key from config file`);
             accessory.context.token = deviceConfig.advanced_options.token;
```

Comment:
When we began to refactor the code to the new structure (where we have the token/key in the config.json) I thought about if we need to store them in the cache or not as they will be available all the time through the config file. What are your thoughts about this?

Reply:
Now that we **require** a devices array, I think it probably is redundant to cache.  When would it ever be used?  Do we have code to remove a cached accessory if it is removed from config.json?

For that matter... we don't have a button in the settings GUI to delete a device.  Oops.  Need to add that.

Reply:
Mind you, we should be cautious about deleting a cached homebridge accessory.  If the device is removed from the config.json just temporarily, then is put back, what is the user expectation?  If we delete the accessory from cache then I think Apple Homekit is going to delete it.... along with any automations connected to it.  And that is nasty. I have had to rebuild a bunch of automations when I tinkered with some Homebridge accessories, it is not fun... and, Apple in their wisdom don't provide a backup/restore for Home automations. What are they thinking?

Reply:
There is a delete button in the GUI by default. It's a little 'x' in the top right corner, when you set the fields for a device.

Currently we don't delete accessories at all I think, cached devices can be deleted by hand using the GUI. I think it should remain as is.

Reply:
You're right, delete 'X' is there.

Right now if you delete a device, but it is still on your network, then on restart the plugin prints a warning in the log that a new device is discovered and that you should add it to config.json.  `AddDevice()` is NOT called, so the cached accessory remains visible in Homebridge, but is inactive.

We could leave it as-is.  Or we could delete the cached accessory.  Or we could call AddDevice... essentialy ignoring the removal from config.json.

If we leave it as-is, then I think we need to document that the user must manually delete the cached accessory in Homebridge.  Deleting the cached accessory in the plugin is bad for reason I stated above.  And I don't think we should simply ignore a user deleting it.  So I think best is to leave the code as-is and update the documentation.

Reply:
All of the plugins I use applied the same methodology that we have talked about, so I think this is the best way. In the other hand we don't use the cache for anything (I think?) so it could be safely deleted.

Reply:
No, I'm wrong, we can't delete the cache for one reason: it contains the UUID, which are being used by Homekit I guess. So yeah, as you said, if we delete it, and later we add it back, the automations won't contain the readded device.

Reply:
I updated the readme, and fixed issue in index.html where the table header titles could be split onto multiple lines when they shouldn't be.  And an we were setting serial number and model into the accessory cache for devices already cached instead of newly added devices!